### PR TITLE
Python PCA API

### PR DESCRIPTION
**Pull request by archanasri**

### CODE REVIEW

File: Testing/ParticlesTests/ParticlesTests.cpp [hunk 1]
```diff
@@ -27,6 +28,23 @@ const std::vector<std::string> filenames = {
   test_dir + "seg.ellipsoid_21.isores.pad.com.aligned.cropped.tpSmoothDT_world.particles"
 };
 
+TEST(ParticlesTests, vnlVector)
```
Comment (cchriste) [inline Testing/ParticlesTests/ParticlesTests.cpp:31]: Verification of results should be added to the test in order to ensure underlying functionality of the component being tested remains consistent.

File: Testing/ParticlesTests/ParticlesTests.cpp [hunk 2]
```diff
@@ -27,6 +28,23 @@ const std::vector<std::string> filenames = {
...
+  ParticleShapeStatistics stats;
+  stats.DoPCA(particleSystem);
+  stats.PrincipalComponentProjections();
+  Eigen::MatrixXd pcaVec = stats.PCALoadings();
+  for(int i=0; i<pcaVec.rows(); i++)
+  {
+    std::cout << "\n";
+    for(int j=0; j<pcaVec.cols(); j++)
+    {
+      std::cout << pcaVec(i,j) << " ";
```
Comment (cchriste) [inline Testing/ParticlesTests/ParticlesTests.cpp:43]: Tests should not print anything to the console.

File: Testing/PythonTests/pca.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  pcaLoadings = pss.pcaLoadings()
+
+  scree = np.array(pss.percentVarByMode())
+  return pcaLoadings, scree
+
+particlesData = readData()
+sklearn_loadings, sklearn_scree = sklearn_pca(particlesData)
+sw_loadings, sw_scree = shapeworks_pca(particlesData)
+
+if np.allclose(sklearn_scree, sw_scree) == False or np.sum(sw_loadings-sw_loadings) != 0.0:
```
Comment (cchriste) [inline Testing/PythonTests/pca.py:40]: As above, test results should be compared against expected output.

**Review by cchriste — CHANGES_REQUESTED** [review]
Good updates overall.
Tests need improvement.

File: Testing/ParticlesTests/ParticlesTests.cpp (hunk 2 above)
Comment (archanasri) [reply Testing/ParticlesTests/ParticlesTests.cpp]: Removed

File: Testing/ParticlesTests/ParticlesTests.cpp (hunk 1 above)
Comment (archanasri) [reply Testing/ParticlesTests/ParticlesTests.cpp]: Done

File: Testing/PythonTests/pca.py (hunk 3 above)
Comment (archanasri) [reply Testing/PythonTests/pca.py]: Done

**Review by cchriste — APPROVED** [review]

**Comment by sheryjoe** [comment]
@iyerkrithika21 can you review this PR?

**Comment by akenmorris** [comment]
Thanks for adding an entry to UseCaseTests!

**Review by akenmorris — APPROVED** [review]
Looks good to me.

**Review by iyerkrithika21 — APPROVED** [review]

**Pull request merged**